pkg/syncmap/syncmap.go

```diff
@@ -0,0 +1,82 @@
+package syncmap
+
+import (
+	"maps"
+	"sync"
+)
+
+// A SyncMap is a map of a string to a generified value which is locked for safe
+// access from multiple threads.
+// It is effectively a generic version of Golang's standard library sync.Map.
+// Admittedly, that has optimizations for multithreading performance that we do
+// not here; thus, SyncMap should not be used in truly performance sensitive
+// areas, but places where code cleanliness is more important than raw
+// performance.
+// SyncMap should always be passed by reference, not by value, to ensure thread
+// safety is maintained.
+type SyncMap[K comparable, V any] struct {
```

Comment:
oh one other thing `syncmap.SyncMap` kinda sucks as type name, maybe we just name it `Map` so it is more similar to `sync.Map`?